Library/Homebrew/resource_auditor.rb

```diff
@@ -20,14 +20,25 @@ def initialize(resource, spec_name, options = {})
       @spec_name = spec_name
       @online    = options[:online]
       @strict    = options[:strict]
+      @only      = options[:only]
+      @except    = options[:except]
       @problems  = []
     end
 
     def audit
-      audit_version
-      audit_download_strategy
-      audit_checksum
-      audit_urls
+      only_audits = @only
+      except_audits = @except
+
+      methods.map(&:to_s).grep(/^audit_/).each do |audit_method_name|
+        name = audit_method_name.delete_prefix("audit_")
+        if only_audits
+          next unless only_audits.include?(name)
```

Comment:
```suggestion
          next if only_audits.exclude?(name)
```

Reply:
I find `unless ... include?` easier to read (because a list of things doesn't exclude things but lists the things that are included) but in this specific case `next if` has the advantage that, if desired, we could rewrite the `if` block as:

```ruby
next if only_audits.present? && only_audits.exclude?(name)
next if except_audits.present? && except_audits.include?(name)
```

Reply:
Thinking about this, we can simplify it further by using the safe navigation operator:

```ruby
next if only_audits&.exclude?(name)
next if except_audits&.include?(name)
```

I've updated the `#audit` method in both classes to use this format.